core/cmake/BareosFindAllLibraries.cmake

```diff
@@ -188,6 +188,33 @@ if(ENABLE_JANSSON)
   find_package(Jansson)
 endif()
 
+option(ENABLE_GRPC "Build with grpc support" OFF)
+set(HAVE_GRPC 0)
+
+if(ENABLE_GRPC)
+  find_package(Protobuf)
+  # Find gRPC installation Looks for gRPCConfig.cmake file installed by gRPC's
+  # cmake installation.
+  find_package(gRPC CONFIG)
+  if(NOT Protobuf_FOUND)
+    message(FATAL_ERROR "ENABLE_GRPC=ON but protobuf not found")
+  elseif (Protobuf_VERSION VERSION_LESS_EQUAL 3.12.0)
+	  message(FATAL_ERROR "PROTOBUF is too old, we need at least 3.12.0 but have only ${Protobuf_VERSION} for optional fields in proto3")
+  elseif(NOT gRPC_FOUND)
+    message(INFO "ENABLE_GRPC=ON but grpc not (yet) found, fallback to pkg_check_modules()")
+    pkg_check_modules (gRPC grpc++ REQUIRED)
+    find_library(rpc++_reflection NAMES grpc++_reflection HINTS /usr/lib/x86_64-linux-gnu/ REQUIRED)
+    find_library(grpc++ NAMES grpc++ HINTS /usr/lib/x86_64-linux-gnu/ REQUIRED)
+    message(WARNING "gRPC_VERSION: ${gRPC_VERSION}")
+    set(HAVE_GRPC 1)
+    if(NOT gRPC_FOUND)
+      message(FATAL_ERROR "ENABLE_GRPC=ON but grpc not found")
+    endif()
+  else()
+    set(HAVE_GRPC 1)
+  endif()
+endif()
```

Comment:
This sucks.
it should look like
```
find_package(Protobuf 3.12.0 REQUIRED)
find_package(gRPC CONFIG REQUIRED)
```
the handling to find gRPC via pkg_check_modules should be moved into a find module. Also `HAVE_GRPC` is useless: you can just check `ENABLE_GRPC` for that information.